content/guides/advanced/full-stack-solana-development.md

```diff
@@ -0,0 +1,1180 @@
+
+```ts
+import { useEffect, useState } from "react";
+import { useConnection } from "@solana/wallet-adapter-react";
+import { program, counterPDA, CounterData } from "../anchor/setup";
+
+export default function CounterState() {
+  const { connection } = useConnection();
+  const [counterData, setCounterData] = useState<CounterData | null>(null);
+
+  useEffect(() => {
+    // Fetch initial account data
+    program.account.counter.fetch(counterPDA).then((data) => {
+      setCounterData(data);
+    });
+
+    // Subscribe to account change
+    const subscriptionId = connection.onAccountChange(
+        // The address of the account we want to watch
+      counterPDA,
+        // callback for when the account changes
+      (accountInfo) => {
+        setCounterData(
+          program.coder.accounts.decode("counter", accountInfo.data)
+        );
+      }
+    );
+
+    return () => {
+      // Unsubscribe from account change
+      connection.removeAccountChangeListener(subscriptionId);
+    };
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [program]);
+
+    // Render the value of the counter
+  return <p className="text-lg">Count: {counterData?.count?.toString()}</p>;
+}
+```
+
+Pretty straight-forward what's happening here. We use
+`program.account.counter.fetch(counterPDA)` to fetch the value of counter on
+load and then subscribe to any changes with `connection.onAccountChange`, which
+takes in a callback as the second value. The callback decodes the data we get
+back and sets it. EZPZ!
+
+The final step is to add the component to our `App.tsx` file. Add this to the
+bottom of the file:
+
+```ts
+// ... previous imports
+// Import the component we just created
+import CounterState from "./components/counter-state";
+
+function App() {
+  const network = WalletAdapterNetwork.Devnet;
+  const endpoint = useMemo(() => clusterApiUrl(network), [network]);
+
+  const wallets = useMemo(
+    () => [new PhantomWalletAdapter()],
+    [network]
+  );
+
+  return (
+    <ConnectionProvider endpoint={endpoint}>
+      <WalletProvider wallets={wallets} autoConnect>
+        <WalletModalProvider>
+          <WalletMultiButton />
+          <h1>Hello Solana</h1>
+          // ADD THIS LINE
+          <CounterState />
+        </WalletModalProvider>
+      </WalletProvider>
+    </ConnectionProvider>
+  );
+}
+
+export default App;
+```
+
+Now if you open up `http://localhost:5173/` and you should see something like:
+
+```
+Count: 2
+```
+
+WOAH. WE JUST READ DATA FROM THE BLOCKCHAIN. Let's keep the momentum rolling by
+writing!
+
+### Writing to the blockchain
+
+Start by adding the `spl-token` library. We'll use this to get the associated
+token account address for the user's wallet.
+
+```
+yarn add @solana/spl-token
+```
+
+The associated token account is where the user's tokens for this program are
+stored. We need this cause we're going to mint a token in the increment
+transaction.
```

Comment:
Whoops, this is from an older version of the guide where I did token minting with a CPI in the increment instruction. I took it out because it was too much but somehow left in this bit.